### PR TITLE
🔄 fix: update navigation logic in `useFocusChatEffect` to ensure correct search parameters are used

### PR DESCRIPTION
**Pull request by mawburn**
## Summary

Fixes an issue with https://github.com/danny-avila/LibreChat/pull/7151 where the params being set were getting clobbered. It looks like something that was updated after I put my PR up and I didn't catch it. 

- https://github.com/danny-avila/LibreChat/pull/7151

This fixes it. Switching models and to agents will update the URL as intended. 

https://github.com/user-attachments/assets/f5e0f81a-02c2-4565-8619-70e3f9f0f40f

## Change Type

Please delete any irrelevant options.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] My changes do not introduce new warnings

## Copilot summary

This pull request enhances the `useFocusChatEffect` hook and adds comprehensive test coverage for its functionality. The changes ensure better handling of URL parameters, focus behavior, and edge cases, particularly in scenarios involving state and device types.

### Enhancements to `useFocusChatEffect` functionality:
* Updated the `navigate` call to prioritize `window.location.search` over `location.search` for URL parameters, ensuring consistency with the browser's actual URL. (`client/src/hooks/Chat/useFocusChatEffect.ts`, [client/src/hooks/Chat/useFocusChatEffect.tsR14-R23](diffhunk://#diff-ca2ac85398b1f9e65e95366e6d65543bed8b6ae155d16c50ca743c6b1602eb54R14-R23))

### Comprehensive test coverage:
* Added a new test suite in `useFocusChatEffect.spec.tsx` to validate the behavior of the hook under various conditions, including:
  - Focusing the textarea based on `location.state.focusChat` and device type.
  - Handling different URL parameter scenarios, such as prioritizing `window.location.search` and managing malformed or special characters in parameters.
  - Testing edge cases like null or undefined search parameters, and when `location.state` or `focusChat` is missing. (`client/src/hooks/Chat/__tests__/useFocusChatEffect.spec.tsx`, [client/src/hooks/Chat/__tests__/useFocusChatEffect.spec.tsxR1-R397](diffhunk://#diff-3028aaabb5191d4a80411655960e993c7dee31a48b5e13d247482df4d29da52cR1-R397))

### CODE REVIEW

**Review by nhtruong — COMMENTED** [review]
Nice Find and this looks good!

File: client/src/hooks/Chat/useFocusChatEffect.ts [hunk 1]
```diff
@@ -11,11 +11,16 @@ export default function useFocusChatEffect(textAreaRef: React.RefObject<HTMLText
...
       /** Check if the device is not a touchscreen */
       if (!window.matchMedia?.('(pointer: coarse)').matches) {
         textAreaRef.current?.focus();
       }
-      navigate(`${location.pathname}${location.search ?? ''}`, { replace: true, state: {} });
+
+      navigate(`${location.pathname}${window.location.search}`, {
+        replace: true,
+        state: {},
+      });
```
Comment (mawburn) [inline client/src/hooks/Chat/useFocusChatEffect.ts:23]: Just to clarify what's going on here. Since I wasn't able to use the react router functions in the atom family in https://github.com/danny-avila/LibreChat/pull/7151, then it wasn't being populated into the react router functions like `const location = useLocation()` and this changes it to pull from the actual `window.location.search` instead, where `location.search` was empty and clobbering everything.

File: client/src/hooks/Chat/useFocusChatEffect.ts [hunk 2]
```diff
@@ -11,11 +11,16 @@ export default function useFocusChatEffect(textAreaRef: React.RefObject<HTMLText
...
       /** Check if the device is not a touchscreen */
       if (!window.matchMedia?.('(pointer: coarse)').matches) {
         textAreaRef.current?.focus();
       }
-      navigate(`${location.pathname}${location.search ?? ''}`, { replace: true, state: {} });
+
+      navigate(`${location.pathname}${window.location.search ?? ''}`, {
+        replace: true,
+        state: {},
+      });
```
Comment (mawburn) [inline client/src/hooks/Chat/useFocusChatEffect.ts:23]: Just to clarify what's going on here. Since I wasn't able to use the react router functions in the atom family in https://github.com/danny-avila/LibreChat/pull/7151, then it wasn't being populated into the react router functions like `const location = useLocation()`. 

So this PR changes it to pull from the actual `window.location.search` instead, where `location.search` was empty and clobbering everything.

**Review by danny-avila — APPROVED** [review]

**Comment by mawburn** [comment]
@danny-avila I forgot to mention this, but this could be why you were seeing a discrepancy here: https://github.com/danny-avila/LibreChat/pull/7151#issuecomment-2858761185

> I tried this last night and noticed https://github.com/danny-avila/LibreChat/issues/6980 was still not resolved with this PR, so I closed that issue in another PR.

Because, I definitely had this working in that branch. I just wasn't checking up on it when I made updates from main like I should have. But, this file I fixed here seems totally unrelated at first glance.

**Comment by danny-avila** [comment]
@mawburn thank you for the PR and all the clarifying notes!
